spec/rubocop/cop/style/mutable_constant_spec.rb

```diff
@@ -43,6 +43,43 @@
     end
   end
 
+  shared_examples 'literals that are frozen' do |o|
+    let(:prefix) { o }
+
+    it_behaves_like 'immutable objects', '[1, 2, 3]'
+    it_behaves_like 'immutable objects', '%w(a b c)'
+    it_behaves_like 'immutable objects', '{ a: 1, b: 2 }'
+    it_behaves_like 'immutable objects', "'str'"
+    it_behaves_like 'immutable objects', '"top#{1 + 2}"'
+    it_behaves_like 'immutable objects', '1'
+    it_behaves_like 'immutable objects', '1.5'
+    it_behaves_like 'immutable objects', ':sym'
+    it_behaves_like 'immutable objects', 'FOO + BAR'
+    it_behaves_like 'immutable objects', 'FOO - BAR'
+    it_behaves_like 'immutable objects', "'foo' + 'bar'"
+    it_behaves_like 'immutable objects', "ENV['foo']"
+    it_behaves_like 'immutable objects', "::ENV['foo']"
+  end
+
+  shared_examples 'literals that are not frozen' do |o|
+    let(:prefix) { o }
+
+    it_behaves_like 'mutable objects', '[1, 2, 3]'
+    it_behaves_like 'mutable objects', '%w(a b c)'
+    it_behaves_like 'mutable objects', '{ a: 1, b: 2 }'
+    it_behaves_like 'mutable objects', "'str'"
+    it_behaves_like 'mutable objects', '"top#{1 + 2}"'
+
+    it_behaves_like 'immutable objects', '1'
+    it_behaves_like 'immutable objects', '1.5'
+    it_behaves_like 'immutable objects', ':sym'
+    it_behaves_like 'immutable objects', 'FOO + BAR'
+    it_behaves_like 'immutable objects', 'FOO - BAR'
+    it_behaves_like 'immutable objects', "'foo' + 'bar'"
+    it_behaves_like 'immutable objects', "ENV['foo']"
+    it_behaves_like 'immutable objects', "::ENV['foo']"
+  end
+
```

Comment:
@marcandre - Made a change here. Was reviewing code and found that 
```ruby
# shareable_constant_value: literal
C = Set[2, 3] # Raises this error "cannot assign unshareable object to C (Ractor::IsolationError)"
```

I think we would need to write a separate cop for shareable_constant_value to handle the real purpose of shareable constant - ractor safe objects.

Reply:
Yes, `Set[]` is not (yet) considered like a "builtin" literal, so it is normal it behaves differently from Array/Hash/...